### PR TITLE
Unify modal dialogs/deposit on transfer screen

### PR DESCRIPTION
**Pull request by taleldayekh**
This PR implements the unified modal design for deposit on the transfer screen.

---
This change is [Reviewable](https://reviewable.io/reviews/raiden-network/light-client/977)

### CODE REVIEW

**Comment by taleldayekh** [comment]
Forgot to adjust the tests accordingly, on it now.

File: raiden-dapp/src/components/ChannelDepositDialog.vue [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  @Prop({ required: true })
+  identifier!: number;
+  @Prop({ required: true })
+  visible!: boolean;
+  @Prop({ required: true })
+  token!: Token;
+  @Prop({ required: true })
+  loading!: boolean;
+  @Prop({})
+  done?: boolean;
```
Comment (kelsos) [inline raiden-dapp/src/components/ChannelDepositDialog.vue]: Would it be possible to make the `@Prop({required:false, default: false})
and `done!: boolean`?

I had a lot of issues and warnings in the past about undefined properties in class syntax components breaking reactivity.

File: raiden-dapp/src/components/ChannelDepositDialog.vue [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  deposit: string = '0.0';
+  valid: boolean = false;
+
+  @Emit()
+  cancel() {}
+
+  mintDeposit() {
```
Comment (kelsos) [inline raiden-dapp/src/components/ChannelDepositDialog.vue]: Is mint the correct term here? you are not `minting` the deposit. You usually have the tokens aquired externally and then you deposit them to the channel.

File: raiden-dapp/src/components/ChannelDepositDialog.vue (hunk 1 above)
Comment (taleldayekh) [reply raiden-dapp/src/components/ChannelDepositDialog.vue]: Absolutely, will do that.

File: raiden-dapp/src/components/ChannelDepositDialog.vue (hunk 2 above)
Comment (taleldayekh) [reply raiden-dapp/src/components/ChannelDepositDialog.vue]: You're right, I'll change it to just `deposit`.

**Comment by taleldayekh** [comment]
Is the failing `test_dapp` related to what @andre mentioned in the dailies? They are all passing when I run the unit tests locally.

**Comment by taleldayekh** [comment]
Is the failing `test_dapp` related to what @andrevmatos mentioned in the dailies? They seemed to pass when I ran the unit tests locally but now I'm having `allocation failure scavenge might not succeed` on the `channel-deposit.spec.ts` test, so not really anything that seems related to the test code itself?

File: raiden-dapp/src/components/ChannelDepositDialog.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  @Prop({ required: false, default: false })
+  done?: boolean;
+
+  deposit: string = '0.0';
+  valid: boolean = false;
+
+  @Emit()
+  cancel() {}
+
+  depositTokens() {
```
Comment (nephix) [inline raiden-dapp/src/components/ChannelDepositDialog.vue]: We may need an `@Emit` here since we check for `emitted.depositTokens` in the tests?

**Comment by nephix** [comment]
Can you please also add a changelog entry

**Comment by nephix** [comment]
Good job, thank you 🙏

**Comment by taleldayekh** [comment]
> looks good thank you

Thanks a lot!

**Pull request merged**